instance/management/commands/migrate_swift_to_s3.py

```diff
@@ -0,0 +1,135 @@
+            authurl=instance.swift_openstack_auth_url,
+            tenant_name=instance.swift_openstack_tenant,
+            auth_version='2',
+            os_options={'region_name': instance.swift_openstack_region}
+        )
+
+    def _create_rclone_config(self, access_key, secret_key):
+        """
+        Creates rclone config, to ease the process of copying
+        """
+        rclone_config = "rclone config create ocim-s3 s3 access_key_id %s secret_access_key %s" % (access_key, secret_key)
+        print("Would run this command: %s.       ← FIXME test it and run it" % rclone_config)
+        return
+
+        subprocess.Popen(rclone_config % (access_key, secret_key), shell=True)
+
+
+    def _copy_subdirectory(self, key_name, new_name, swift_container, s3_bucket):
+        """
+        Copy subdirectory using rclone
+        """
+        LOG.info("%s ---> %s" % (key_name, new_name))
+        command = "rclone copy ocim-swift:%s/%s ocim-s3:%s/%s" % (swift_container, key_name, s3_bucket, new_name)
+        print("Would run this command: %s.       ← FIXME test it and run it" % command)
+        return
+        subprocess.Popen(command, shell=True)
+
+    def _migrate_swift_to_s3(self, instance):
+        # FIXME Will these 2 values already be stored in the instance? If not, where to get them from? (Or should we set them and save?)
+        # FIXME and in particular: will the S3 region match the SWIFT region? (some clients don't want their data in a particular region; see comments on ticket)
+        self._create_rclone_config(instance.s3_access_key, instance.s3_secret_access_key)
+
+        copied = []  # FIXME describe variable
+        swift = self._get_swift_connection(instance)
+
+        container = instance.swift_container_name
+        s3_bucket = instance.s3_bucket_name  # FIXME will this instance already have the right bucket name? If not, we must set it and save the instance
+        LOG.info("Migrating container: %s" % container)
+        for item in swift.get_container(container)[1]:
+            LOG.info("Seeing file: %s", item['name'])
+
+            # FIXME test this part; will it also copy files in the root? (in addition to directories):
+            # get only directories
+            key_name = re.search('(.*\/)', item['name']).group(1)
+            if key_name in copied:
+                continue
+
+            # replace new name if they contain submissions_attachments (there is a bug on swift implementation that makes
+            # files be uploaded to the wrong directory)
+            # The changes to do are like:
+            #   submissions_attachmentsbadges/ ---> submissions_attachments/badges/
+            #   submissions_attachmentsuser_tasks ---> submissions_attachments/user_tasks/
+            # etc.
+            # FIXME what about submissions_attachmentssubmissions_attachments? In OC-4395 we needed: rclone-sync someserver/submissions_attachmentssubmissions_attachments/ someserver/submissions_attachments/
```

Comment:
@clemente not sure what you mean with this comment, could you explain a bit more?

Reply:
I meant the „submissions_attachment“ directory itself (or its contents), it seems it's a special case (at least in Jill's example it was a special case requiring an extra line). Maybe the files inside `submissions_attachmentssubmissions_attachments` must be moved to `submissions_attachments`.
I should check it.

Reply:
Good catch! I missed that, and it seems yes, it was a special case, and looking at the command she run it seems they have to be moved to just submissions_attachments/